### PR TITLE
ISSUE #545 updated bouncer worker to use yup

### LINKED ISSUE #545: use yup for config parsing
We currently do not validate the config and are plugging in default values by hand.

If we utilise Yup it will make this process more robust and the code easier to read.

### PR DESCRIPTION
**Pull request by sebjf**
This fixes #545

#### Description
This PR updates `config.js` in `bouncer_worker` so that it uses `yup` to perform basic transforms and validation of the config, instead of us doing it by hand.

The yup schema has been written to match the Readme (which has also had a couple of outdated segments removed), for the parts of the schema that are used by bouncer worker.

The log directories fallback is still done as a post processing step - it is ambigious whether the `default` transformer of yup can use another field as a fallback; the answers online don't seem to work with our version, and its complicated by the fact the fields are not siblings either. For simplicitly, these two fallbacks remain conditionals, with yup validating the types of the fields involved.

Finally, a few packages flagged by dependabot have been updated.

### CODE REVIEW

File: tools/bouncer_worker/src/lib/config.js [hunk 1]
```diff
@@ -18,57 +18,101 @@
...
+});
+
+const db = object({
+	dbhost: string(),
+	dbport: number().min(1).max(65536),
+});
+
+const schema = object({
+	timeoutMS: number().default(180 * 60 * 1000),
+	logLocation: string(),
```
Comment (carmenfan) [inline tools/bouncer_worker/src/lib/config.js:73]: I think logLocation is just old place where we put workerLogPath?

I think we can probably get rid of it

File: tools/bouncer_worker/src/lib/config.js [hunk 2]
```diff
@@ -18,57 +18,101 @@
...
 
 // eslint-disable-next-line consistent-return
 const init = () => {
 	try {
 		Config.configPath = params.config || path.resolve(__dirname, '../../config.json');
-		const config = JSON.parse(fs.readFileSync(Config.configPath));
-		applyDefaultValuesIfUndefined(config);
+		let config = JSON.parse(fs.readFileSync(Config.configPath));
+		config = schema.cast(config);
+		config = schema.validateSync(config);
```
[the remark this answers is not in the document]
Comment (carmenfan) [reply tools/bouncer_worker/src/lib/config.js:89]: can we turn on the option of `stripUnknown`? so whoever wants to add an option will have to make sure they add it in the schema

**Review by carmenfan — CHANGES_REQUESTED** [review]

File: tools/bouncer_worker/src/lib/config.js [hunk 3]
```diff
@@ -18,57 +18,101 @@
...
-	config.processMonitoring.memoryIntervalMS = config.processMonitoring.memoryIntervalMS || 100;
-
-	// create connection string for db
-	if (!config.db.connectionString) {
-		config.db.connectionString = `mongodb://${config.db.dbhost}:${config.db.dbport}`;
-	}
+const rabbitmq = object({
+	host: string().required(),
+	sharedDir: string().required(),
+	worker_queue: string(),
```
[the remark this answers is not in the document]
Comment (sebjf) [reply tools/bouncer_worker/src/lib/config.js:30]: I think so. These [conditionally](https://github.com/3drepo/3drepobouncer/blob/5f559069381254d623f00875d424769e2b297c1f/tools/bouncer_worker/src/lib/queueHandler.js#L40) populate the `queueHandlers` object, but this object is simply used as a lookup when the actual queues to listen to are enumerated from the arguments, so there should be no harm in defining them.

File: tools/bouncer_worker/src/lib/config.js [hunk 4]
```diff
@@ -18,57 +18,101 @@
...
-		const config = JSON.parse(fs.readFileSync(Config.configPath));
-		applyDefaultValuesIfUndefined(config);
+		let config = JSON.parse(fs.readFileSync(Config.configPath));
+		config = schema.cast(config);
+		config = schema.validateSync(config);
+
+		// If a license is set, we should specify a unique Id for this worker.
+		// Always initialise a new one, so there is no confusion between instances.
+		if (config.repoLicense) {
+			config.instanceId = uuidv4();
```
[the remark this answers is not in the document]
Comment (carmenfan) [reply tools/bouncer_worker/src/lib/config.js:94]: Ah I see what you mean... in which case we could leave it as is.

**Review by carmenfan — APPROVED** [review]

**Pull request merged**